File: Assignment #2/Codes/Python/3.py
def findSum(N,firstElement,interval):
	if(N == 0):
		return 0
	if(N == 1):
		return firstElement
	else:
		return findSum(N-1,interval,firstElement) + firstElement + (N-1) * interval

#Main

t = int(input('How many times?'))
for i in range(t):
	print('Iteration: ', i+1)
	n = int(input('N : '))
	f = int(input('First Element : '))
	i = int(input('Interval : '))
	print('Sum is : ', findSum(n,f,i))
	print('\n')

File: Assignment #2/Codes/Prolog/3.pl
sum(1,FirstElement,_,FirstElement):- !.
sum(N,FirstElement,Interval,Result):-
	N1 is N-1,
	sum(N1,FirstElement,Interval,TemporaryResult),
	Result is TemporaryResult + FirstElement + (N - 1) * Interval.